agent/provider/src/execution/exeunit_instance.rs

```diff
@@ -96,6 +96,9 @@ impl ExeUnitInstance {
             })?;
 
         let output = child.wait_with_output().await?;
+        if !output.status.success() {
+            anyhow::bail!(String::from_utf8_lossy(output.stderr.as_slice()).to_string())
+        }
```

Comment:
This function is used only by `test` and `offer-template`, so returning here with an error will not break anything (`wasm` runtime uses `--help` arg to not return an Error)